src/System.IO.Compression.Brotli/System/IO/Compression/Brotli/BrotliPrimitives.cs

```diff
@@ -16,41 +17,82 @@ public struct BrotliPrimitives
         const int defQuality = 11;
         const int defLgWin = 24;
 
-        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out nuint bytesConsumed, out nuint bytesWritten)
+        public static int BrotliEncoderMaxCompressedSize(int input_size)
+        {
+            int num_large_blocks = input_size >> 24;
+            int tail = input_size - (num_large_blocks << 24);
```

Comment:
I believe it is faster. One is an and operation and the other is a subtraction and bit shift.